web/frontend/src/components/navigation/index.svelte

```diff
@@ -0,0 +1,226 @@
+<svelte:options immutable />
+
+<script setup lang="ts">
+
+import 'maplibre-gl/dist/maplibre-gl.css';
+import { toast } from '@/lib/toast';
+import { Client, robotApi, navigationApi, type ServiceError, type ResponseStream } from '@viamrobotics/sdk';
+import Collapse from '../collapse.svelte';
+import maplibregl from 'maplibre-gl'; 
+import { setMode, setWaypoint, getWaypoints, removeWaypoint, getLocation } from '@/api/navigation';
+
+export let name: string
+export let client: Client
+export let statusStream: ResponseStream<robotApi.StreamStatusResponse> | null
+
+let updateWaypointsId: number;
+let updateLocationsId: number;
+
+let refreshRate = 500;
+
+const setNavigationMode = async (event: CustomEvent) => {
+  const mode = event.detail.value.toLowerCase() as 'manual' | 'waypoint'
+
+  let pbMode: 0 | 1 | 2 = navigationApi.Mode.MODE_UNSPECIFIED;
```

Comment:
Just for my understanding - is there a reason you are using the raw number values instead of navigationApi.Mode enum for the type?

Reply:
Nope, this is just leftover old code, but I'll fix now :)

Reply:
Updated!